presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationModule.java

```diff
@@ -30,5 +34,31 @@ protected void setup(Binder binder)
             config.setKeyStorePath(internalCommunicationConfig.getKeyStorePath());
             config.setKeyStorePassword(internalCommunicationConfig.getKeyStorePassword());
         });
+
+        install(installModuleIf(InternalCommunicationConfig.class, InternalCommunicationConfig::isKerberosEnabled, kerberosInternalCommunicationModule()));
+    }
+
+    private Module kerberosInternalCommunicationModule()
+    {
+        return binder -> {
+            KerberosInternalCommunicationConfig internalKerberosConfig = buildConfigObject(KerberosInternalCommunicationConfig.class);
+            com.facebook.presto.server.security.KerberosConfig serverKerberosConfig = buildConfigObject(com.facebook.presto.server.security.KerberosConfig.class);
+
+            checkState(serverKerberosConfig.getKeytab() != null && serverKerberosConfig.getKeytab().getAbsolutePath().equals(internalKerberosConfig.getKerberosKeytab().getAbsolutePath()), "kerberos keytab values must match");
```

Comment:
Since we are doing these checks, do we actually need 2 type of properties (for the client and for the server). Can we just use these properties from the server config?